clippy_lints/src/matches.rs

```diff
@@ -539,6 +539,16 @@ declare_clippy_lint! {
     ///     Quz => quz(),
     /// }
     /// ```
+    ///
+    /// If lint is triggered during development because of multiple `todo!` branches,
```

Comment:
I wouldn't add this note to the lint documentation, since I don't see this as an actual issue with this lint.